docs/pages/machine-id/access-guides/kubernetes.mdx

```diff
@@ -21,238 +21,152 @@ controls that Teleport provides for human users.
-Machine ID bot, create one as follows.
+With the appropriate RoleBinding configured in Kubernetes to grant access to
+a specific group, you now need to add this group to the role that the bot
+will impersonate when producing credentials. You also need to grant the bot
+access through Teleport to the cluster itself.
 
-Kubernetes clusters often come with pre-made roles which may meet the needs of
-your application, such as `view` and `edit`. You may view these preexisting
-rules by running the following:
-```code
-$ kubectl get clusterrole
-```
+If you have followed an access guide, you will have created a role and granted
+the bot the ability to impersonate it already. This role just needs to have the
+additional privileges added to it.
 
-If you'd prefer to create a role of your own, the following command can be used
-to create a rule with narrow access to certain resources types in the
-`example-namespace` namespace:
-```code
-## This creates a role allowing `get` and `list` on pods and deployments.
-$ kubectl create role example-role \
-  --namespace=example-namespace \
-  --verb=get,list \
-  --resource=pods,deployments
+Use `tctl edit role/example-bot` to add the following rule to the Teleport role:
+
+```yaml
+spec:
+  allow:
+    kubernetes_labels:
+      '*': '*'
+    kubernetes_resources:
+    - kind: pod
+      namespace: "*"
+      name: "*"
+    kubernetes_groups:
+    - editor
 ```
 
-Next, create a role binding to connect the bot to your role. You may bind either
-the bot's mapped `User` or mapped `Group` to a role, however it must have been
-granted the matching `User` or `Group` via a Teleport role above:
+Adjust the `allow` field for your environment:
 
-```code
-## This binds the new role to the group `example`
-$ kubectl create rolebinding example-rolebinding-group \
-  --namespace=demo \
-  --role=example-role \
-  --group=example
-## Alternatively, this binds the new role to the user `alice`
-$ kubectl create rolebinding example-rolebinding-user \
-  --namespace=demo \
-  --role=example-role \
-  --user=alice
-## If binding to a ClusterRole like `view`, instead run the following:
-$ kubectl create rolebinding example-rolebinding-user \
-  --namespace=demo \
-  --clusterrole=view \
-  --user=alice
-```
+- `kubernetes_labels` should be adjusted to grant access to only the clusters
+  that the bot will need to access. The value shown, `'*': '**'` will grant
+  access to all Kubernetes clusters.
 
-You can grant cluster-wide access to your bot user by instead creating `ClusterRole` and
-`ClusterRoleBinding` resources, which are not namespaced and use an otherwise
-identical syntax.
-</Details>
+- Each item in `kubernetes_resources` must match the namespace and name of a
+  Kubernetes resource that you would like users with `machine-id-kube-role` to
+  access. Currently, Teleport only supports the `pod` kind. In the
+  configuration above, this role can access all pods in all namespaces.
```

Comment:
We'll want to update this since `kubernetes_resources` supports more kinds now (or we can remove the instructions if we're making it a prerequisite to create a role for Kubernetes access).